src/libraries/Microsoft.PowerFx.Core/Public/Values/FormulaValueNewTable.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Public.Types;
+
+namespace Microsoft.PowerFx.Core.Public.Values
+{
+    // Marshalling a tables. 
+    // Tables need to know their record type.
+    // - a ITypeMarshaller  (which can be obtained from a TypeMarshallerCache and a T)
+    // - an explicit RecordType
+    public partial class FormulaValue
+    {
+        #region Host Tables API
+
+        /// <summary>
+        /// Construct a table from records. Assumed that Records must be the same type. 
+        /// Already having RecordValues (as oppossed to a unknown T or errors) lets us avoid type marshalling.
+        /// </summary>
+        /// <param name="values"></param>
+        /// <returns></returns>
+        public static TableValue NewTable(RecordType recordType, params RecordValue[] values)
+        {
+            return NewTable(recordType, (IEnumerable<RecordValue>)values);
+        }
+
+        public static TableValue NewTable(RecordType recordType, IEnumerable<RecordValue> records)
+        {
+            var tableType = recordType.ToTable();
+            return new RecordsOnlyTableValue(IRContext.NotInSource(tableType), records);
+        }
+
+        /// <summary>
+        /// Convenience method to create a table over an array of primitives. 
+        /// The Table will have 1 column 'Value'.
+        /// </summary>
+        /// <typeparam name="T">type of the primitive. This maps directly to a formula Value.</typeparam>
+        /// <param name="values"></param>
+        /// <returns></returns>
+        public static TableValue NewSingleColumnTable<T>(params PrimitiveValue<T>[] values)
+        {
+            IEnumerable<PrimitiveValue<T>> list = values;
+            return NewSingleColumnTable(list);
+        }
+
+        public static TableValue NewSingleColumnTable<T>(IEnumerable<PrimitiveValue<T>> values)
+        {
+            // This is a convenience method. For anything requiring more control,
+            // use NewTable and be explicit. 
+            if (!BuiltinFormulaTypeConversions.TryGetFormulaType(typeof(T), out var fxType))
+            {
+                throw new InvalidOperationException($"Use NewTable() instead");
+            }
+
+            const string valueField = "Value";
```

Comment:
>"Value" [](http://example.com/codeflow?start=38&length=7)

[minor] this constant is used everywhere, we should define it as a proper `const` and reuse it. Can be done in another PR, though